actors/builtin/miner/miner_actor.go

```diff
@@ -188,44 +186,45 @@ func (a Actor) ChangeWorkerAddress(rt Runtime, params *ChangeWorkerAddressParams
 	}
 
 	var st State
-	isWorkerChange := false
 	rt.State().Transaction(&st, func() {
 		info := getMinerInfo(rt, &st)
 
 		// Only the Owner is allowed to change the newWorker and control addresses.
 		rt.ValidateImmediateCallerIs(info.Owner)
 
-		{
-			// save the new control addresses
-			info.ControlAddresses = controlAddrs
-		}
+		// save the new control addresses
+		info.ControlAddresses = controlAddrs
 
-		{
-			// save newWorker addr key change request
-			// This may replace another pending key change.
-			if newWorker != info.Worker {
-				isWorkerChange = true
-				effectiveEpoch = rt.CurrEpoch() + WorkerKeyChangeDelay
-
-				info.PendingWorkerKey = &WorkerKeyChange{
-					NewWorker:   newWorker,
-					EffectiveAt: effectiveEpoch,
-				}
+		// save newWorker addr key change request
+		// This may replace another pending key change.
+		if newWorker != info.Worker {
+			info.PendingWorkerKey = &WorkerKeyChange{
+				NewWorker:   newWorker,
+				EffectiveAt: rt.CurrEpoch() + WorkerKeyChangeDelay,
```

Comment:
ack, we should simply ensure that miners can easily determine the `EffectiveAt` Epoch when staging the change, given this potentially has security/operational implications for them.

Reply:
Just to clarify the existing situation:
The actual delay experienced depends on when the change happens in the deadline. It will be bounded by `WorkerKeyChangeDelay` and `WorkerKeyChangeDelay+WPoStChallengeWindow-1` or `900 - 951` epochs with current parameters. The `ConfirmUpdateWorkerKey` method gives the miner's owner the option of lowering reducing the upper bound to `WorkerKeyChangeDelay` (900) if they don't want to wait for deadline end.

If we quantize `EffectiveAt` there's no use for `ConfirmUpdateWorkerKey`. The timing of the change will be determined solely by the time the change is made within the deadline cycle within the 900-951 range. Is this what we want?

Reply:
I expect current behavior is better...

Reply:
That makes sense. I would be fine with either approach.

Reply:
I think the current approach is better. Quantizing wouldn't make the change happen any sooner, it would just mean that the epoch recorded is the epoch that the change will be automatically applied. This can not be sooner than the "ideal" earliest epoch.